pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -642,6 +642,43 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 			cdLog.Debug("not creating new provision since the deployment is set to try install only once")
 			return reconcile.Result{}, nil
 		}
+		if cd.Spec.InstallAttemptsLimit != nil && cd.Status.InstallRestarts >= int(*cd.Spec.InstallAttemptsLimit) {
+			cdLog.Debug("not creating new provision since the install attempts limit has been reached")
+			conditions, changed := controllerutils.SetClusterDeploymentConditionWithChangeCheck(
+				cd.Status.Conditions,
+				hivev1.ProvisionStoppedCondition,
```

Comment:
We should set this condition when provisioning is stopped due to the try-once annotation as well.